File: laodazi-fix-quotes/fix-quotes.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
中文引号修正脚本
功能：
1. 将所有英文双引号 " " 替换为中文双引号 ""
2. 将所有英文单引号 ' ' 替换为中文单引号 ''

使用方法：
- 读取模式：python3 scripts/fix-quotes.py read <文件路径>
- 写入模式：python3 scripts/fix-quotes.py write <文件路径>
"""

import sys

# 定义中文引号的Unicode编码
LEFT_DOUBLE = '\u201c'   # 中文左双引号 "
RIGHT_DOUBLE = '\u201d'  # 中文右双引号 "
LEFT_SINGLE = '\u2018'   # 中文左单引号 '
RIGHT_SINGLE = '\u2019'  # 中文右单引号 '

def fix_quotes(text):
    """修正文本中的引号"""
    result = []
    double_open = False

    for char in text:
        if char == '"':  # ASCII双引号
            if double_open:
                result.append(RIGHT_DOUBLE)
                double_open = False
            else:
                result.append(LEFT_DOUBLE)
                double_open = True
        elif char == "'":  # ASCII单引号
            # 简单处理：所有英文单引号替换为中文左单引号
            result.append(LEFT_SINGLE)
        else:
            result.append(char)

    return ''.join(result)

def main():
    if len(sys.argv) < 3:
        print("使用方法：")
        print("  读取模式：python3 scripts/fix-quotes.py read <文件路径>")
        print("  写入模式：python3 scripts/fix-quotes.py write <文件路径>")
        sys.exit(1)

    mode = sys.argv[1]
    file_path = sys.argv[2]

    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()

        fixed_content = fix_quotes(content)

        if mode == 'read':
            print(fixed_content)
        elif mode == 'write':
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(fixed_content)
            double_count = content.count('"')
            single_count = content.count("'")
            print(f"✅ 已修正引号：{file_path}")
            print(f"   英文双引号: {double_count} 个 → 中文双引号")
            print(f"   英文单引号: {single_count} 个 → 中文单引号")
        else:
            print(f"❌ 未知模式：{mode}")
            print("支持的模式：read、write")
            sys.exit(1)

    except FileNotFoundError:
        print(f"❌ 文件不存在：{file_path}")
        sys.exit(1)
    except Exception as e:
        print(f"❌ 错误：{e}")
        sys.exit(1)

if __name__ == '__main__':
    main()

File: laodazi-history-writer/scripts/fix-quotes.ts
#!/usr/bin/env bun
/**
 * 中文引号修正脚本
 *
 * 使用方法：
 * - 读取模式：bun run scripts/fix-quotes.ts read <文件路径>
 * - 写入模式：bun run scripts/fix-quotes.ts write <文件路径>
 */

import { readFileSync, writeFileSync } from 'fs';

// 使用Unicode转义序列
const CHINESE_QUOTES = {
  LEFT_DOUBLE: '\u201c',   // 左双引号
  RIGHT_DOUBLE: '\u201d',  // 右双引号
  LEFT_SINGLE: '\u2018',   // 左单引号
  RIGHT_SINGLE: '\u2019',  // 右单引号
};

/**
 * 修正文本中的引号 - 智能判断左右引号
 */
function fixQuotes(text: string): string {
  const result: string[] = [];
  let doubleQuoteOpen = false;
  let singleQuoteOpen = false;

  for (let i = 0; i < text.length; i++) {
    const char = text[i];

    if (char === '"') {
      // 英文双引号替换为中文双引号
      result.push(doubleQuoteOpen ? CHINESE_QUOTES.RIGHT_DOUBLE : CHINESE_QUOTES.LEFT_DOUBLE);
      doubleQuoteOpen = !doubleQuoteOpen;
    } else if (char === "'") {
      // 英文单引号替换为中文单引号
      result.push(singleQuoteOpen ? CHINESE_QUOTES.RIGHT_SINGLE : CHINESE_QUOTES.LEFT_SINGLE);
      singleQuoteOpen = !singleQuoteOpen;
    } else {
      result.push(char);
    }
  }

  return result.join('');
}

/**
 * 读取文件并修正引号
 */
function readFileAndFix(filePath: string): string {
  const buffer = readFileSync(filePath);
  const decoder = new TextDecoder('utf-8');
  const content = decoder.decode(buffer);
  return fixQuotes(content);
}

/**
 * 将修正后的内容写入文件
 */
function writeFileAndFix(filePath: string): void {
  const buffer = readFileSync(filePath);
  const decoder = new TextDecoder('utf-8');
  const content = decoder.decode(buffer);

  const fixed = fixQuotes(content);

  const encoder = new TextEncoder();
  const encoded = encoder.encode(fixed);
  writeFileSync(filePath, Buffer.from(encoded));
}

/**
 * 主函数
 */
function main() {
  const args = process.argv.slice(2);

  if (args.length < 2) {
    console.error('使用方法：');
    console.error('  读取模式：bun run scripts/fix-quotes.ts read <文件路径>');
    console.error('  写入模式：bun run scripts/fix-quotes.ts write <文件路径>');
    process.exit(1);
  }

  const mode = args[0];
  const filePath = args[1];

  if (mode === 'read') {
    const content = readFileAndFix(filePath);
    console.log(content);
  } else if (mode === 'write') {
    writeFileAndFix(filePath);
    console.log(`✅ 已修正引号：${filePath}`);
  } else {
    console.error(`❌ 未知模式：${mode}`);
    console.error('支持的模式：read、write');
    process.exit(1);
  }
}

// 运行主函数
main();
